### ISSUE TITLE
Support for Immutable.js

### ISSUE THREAD

**Issue by tlightsky** [fcomb/redux-logger#7]
transform Immutable.js object to pure json data,before log it out

**Comment by theaqua** [comment]
Can you explain use case? Do you have Immutable.js object (but Immutable.js doesn't have Object, only List and Map) and you want log JSON instead of Immutable List/Map?

**Comment by tlightsky** [comment]
yes,the Immutable.js log info is very hard to read in chrome console log

**Comment by tlightsky** [comment]
I think just add a .toString() to Immutable.js object would work

**Comment by theaqua** [comment]
Hey, I released 1.0.0 without this feature, but will release in next release. Not enough time to investigate & understand `Immutable.js`.

**Comment by idolize** [comment]
@tlightsky I would say `toJS()` would be better than `toString()` for Immutable.js objects (that way you could still use the little dropdown inspector widget), although `toJS()` is pretty specific for Immutable.js, while `toString()` is more universal.

**Comment by tlightsky** [comment]
@idolize yes,toJS() or toJSON is much better :)

**Comment by tlightsky** [comment]
```javascript
import Immutable, { Map, List } from 'immutable';

    if (Map.isMap(state) || List.isList(state)) {
      state = Immutable.toJS(state);
    }
```

**Comment by idolize** [comment]
Actually all the Immutable objects extend `Iterable` so you could just do
`Iterable.isIterable()`.

> "The Iterable is a set of (key, value) entries which can be iterated, and
is the base class for all collections in immutable"

**Comment by idolize** [comment]
Actually all the Immutable objects extend `Iterable` so you could just do `Iterable.isIterable()` :).

> "The Iterable is a set of (key, value) entries which can be iterated, and is the base class for all collections in immutable"

**Comment by farin** [comment]
I think the best solution will be provide option to declare transform function for state.
Then everybody can choose if he wants to log it as using toString or toJS.

It can be also useful if you want log only some part of state.

**Comment by theaqua** [comment]
Yeah, @farin's PR is very helpful.
Released 1.0.5.
https://github.com/fcomb/redux-logger/releases/tag/1.0.5
